staging/src/k8s.io/apiserver/pkg/server/config_test.go

```diff
@@ -32,6 +32,56 @@ import (
 	"k8s.io/client-go/rest"
 )
 
+func TestAuthorizeClientBearerTokenNoops(t *testing.T) {
+	// All of these should do nothing (not panic, no side-effects)
+	cfgGens := []func() *rest.Config{
+		func() *rest.Config { return nil },
+		func() *rest.Config { return &rest.Config{} },
+		func() *rest.Config { return &rest.Config{BearerToken: "mu"} },
+	}
+	authcGens := []func() *AuthenticationInfo{
+		func() *AuthenticationInfo { return nil },
+		func() *AuthenticationInfo { return &AuthenticationInfo{} },
+	}
+	authzGens := []func() *AuthorizationInfo{
+		func() *AuthorizationInfo { return nil },
+		func() *AuthorizationInfo { return &AuthorizationInfo{} },
+	}
+	for _, cfgGen := range cfgGens {
+		for _, authcGen := range authcGens {
+			for _, authzGen := range authzGens {
+				pConfig := cfgGen()
+				pAuthc := authcGen()
+				pAuthz := authzGen()
+				AuthorizeClientBearerToken(pConfig, pAuthc, pAuthz)
+				if pAuthc != nil && !AuthenticationInfoEq(*pAuthc, *authcGen()) {
```

Comment:
This is the same as:
```go
if got, want := authcGen(), pAuthc; !cmp.Equal(want, got) {
  t.Errorf("unexpected diff in AuthenticationConfig: %v", cmp.Diff(want, got))
}
if got, want := authzGen(), pAuthz; !cmp.Equal(want, got) {
  t.Errorf("unexpected diff in AuthorizationConfig: %v", cmp.Diff(want, got))
}
```

Reply:
I don't know what `cmp` is, but I updated the PR to follow the rest of the suggested pattern.

Reply:
It's like refelect.DeepEqual with extra flexibility that's useful for testing like a nice diff output:

https://github.com/kubernetes/kubernetes/blob/1b259bf75a29bab3966d08ec9878efc67a03f9d1/vendor/github.com/google/go-cmp/cmp/compare.go#L40-L82
https://github.com/kubernetes/kubernetes/blob/1b259bf75a29bab3966d08ec9878efc67a03f9d1/staging/src/k8s.io/apimachinery/pkg/util/diff/diff.go

Reply:
I considered using reflect.DeepEqual but decided against it because I want a stricter check, not a more liberal one.  The goal here is just to check that nothing was done.

Reply:
I feel like deep equal is good enough for the purposes of this test. If you hard code a comparison function, it's going to go stale as the config evolves.

Reply:
OK, switched to using `reflect.DeepEqual`.

Reply:
Actually, because of all the `nil`s in the originals, `reflect.DeepEqual` is no more liberal than I want.